### PR TITLE
Oneclick kerberos product tests

### PR DESCRIPTION
**Pull request by arhimondr**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,10 +45,11 @@
         <air.java.version>1.8.0-40</air.java.version>
 
         <dep.antlr.version>4.5.1</dep.antlr.version>
-        <dep.airlift.version>0.125</dep.airlift.version>
+        <dep.airlift.version>0.127-SNAPSHOT</dep.airlift.version>
         <dep.packaging.version>${dep.airlift.version}</dep.packaging.version>
         <dep.slice.version>0.19</dep.slice.version>
         <dep.aws-sdk.version>1.9.40</dep.aws-sdk.version>
+        <dep.tempto.version>1.7-SNAPSHOT</dep.tempto.version>
```
Comment (kokosing) [inline pom.xml:52]: it would be nice to use released versions

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePageSink.java [hunk 2]
```diff
@@ -132,8 +132,11 @@ private static long writeTestFile(HiveClientConfig config, HiveMetastore metasto
                 LineItemColumn column = columns.get(i);
                 BlockBuilder blockBuilder = pageBuilder.getBlockBuilder(i);
                 switch (column.getType()) {
-                    case BIGINT:
```
Comment (kokosing) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHivePageSink.java:135]: no more support for BIGINT?

File: presto-product-tests/conf/docker/singlenode-kerberized/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  presto-master:
+    hostname: presto-master
+    # cdh5-hive-kerberized contains keytabs and the `krb5.conf` within
+    image: 'teradatalabs/cdh5-hive-kerberized:2.6.0-cdh5.6.0-sshd-4e54a19'
+    depends_on:
+      - 'hadoop-master'
+    command: /docker/volumes/presto/presto-product-tests/conf/docker/files/presto-launcher-wrapper.sh singlenode-kerberized run
+    volumes:
+      - ../../../../:/docker/volumes/presto
+      - ../../../../presto-product-tests/conf/presto/etc/hive-kerberized.properties:/docker/volumes/presto/presto-product-tests/conf/presto/etc/catalog/hive.properties
```
Comment (kokosing) [inline presto-product-tests/conf/docker/singlenode-kerberized/docker-compose.yml:24]: is it needed. It looks like it is included in above volume

**Comment by kokosing** [comment]
lgtm % comments

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHivePageSink.java (hunk 2 above)
Comment (arhimondr) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHivePageSink.java:135]: Now `BIGINT` columns in tpch are called `IDENTIFICATOR`

File: presto-product-tests/conf/docker/singlenode-kerberized/docker-compose.yml (hunk 3 above)
Comment (arhimondr) [inline presto-product-tests/conf/docker/singlenode-kerberized/docker-compose.yml:24]: In such way i override the default hive configuration with the kerberized one.

File: pom.xml (hunk 1 above)
Comment (arhimondr) [inline pom.xml:52]: `Airlift` is still `SNAPSHOT` anyway. Once `Airlift` is released we can release `Tempto` as well.

File: presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java [hunk 4]
```diff
@@ -130,7 +129,7 @@ public static Page createInputPage()
             DOUBLE.writeDouble(pageBuilder.getBlockBuilder(EXTENDED_PRICE), lineItem.getExtendedPrice());
             DOUBLE.writeDouble(pageBuilder.getBlockBuilder(DISCOUNT), lineItem.getDiscount());
             DATE.writeLong(pageBuilder.getBlockBuilder(SHIP_DATE), lineItem.getShipDate());
-            BIGINT.writeLong(pageBuilder.getBlockBuilder(QUANTITY), lineItem.getQuantity());
+            DOUBLE.writeDouble(pageBuilder.getBlockBuilder(QUANTITY), lineItem.getQuantity());
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/sql/gen/BenchmarkPageProcessor.java]: this changes benchmark semantics, isn't it? Why not IDENTIFICATOR

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 5]
```diff
@@ -772,7 +772,7 @@ public void testDistinctMultipleFields()
     public void testDistinctJoin()
             throws Exception
     {
-        assertQuery("SELECT COUNT(DISTINCT b.quantity), a.orderstatus " +
+        assertQuery("SELECT COUNT(DISTINCT CAST(b.quantity AS BIGINT)), a.orderstatus " +
```
Comment (sopel39) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java:775]: is cast needed?

File: pom.xml [hunk 6]
```diff
@@ -459,7 +459,7 @@
             <dependency>
                 <groupId>io.airlift.tpch</groupId>
                 <artifactId>tpch</artifactId>
-                <version>0.5</version>
+                <version>0.6</version>
```
Comment (sopel39) [inline pom.xml:463]: why this change?

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java (hunk 5 above)
Comment (arhimondr) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java:775]: This commit is just a partial cherry pick of https://github.com/prestodb/presto/commit/dbbe14e7f974b80aba11b4a45695ec81e84545e2 from master. I'm not quite sure what exactly changes were introduced with TPCH 0.6.

File: pom.xml (hunk 6 above)
Comment (arhimondr) [inline pom.xml:463]: Because newer Tempto version is dependent on Tpch 0.6. This migration commit is already on master. I think you shouldn't review it. The very next sprint branch will aready have this commit from master.

File: presto-product-tests/conf/docker/singlenode-kerberized/docker-compose.yml (hunk 3 above)
Comment (sopel39) [inline presto-product-tests/conf/docker/singlenode-kerberized/docker-compose.yml]: why you need to override hive.properties on presto-master machine?

File: presto-cli/src/main/java/com/facebook/presto/cli/ClientOptions.java [hunk 7]
```diff
@@ -145,7 +148,7 @@ public KerberosConfig toKerberosConfig()
         if (krb5CredentialCachePath != null) {
             config.setCredentialCache(new File(krb5CredentialCachePath));
         }
-
+        config.setUseCanonicalHostname(!krb5DisableRemoteServiceHostnameCanonicalization);
```
Comment (sopel39) [inline presto-cli/src/main/java/com/facebook/presto/cli/ClientOptions.java:151]: This wasn't needed before. Why it is needed now?
